src/core/cdr/src/dds_cdrstream.c

```diff
@@ -1810,9 +1817,76 @@ static const uint32_t *dds_stream_skip_adr_default (uint32_t insn, char * __rest
       break;
     }
   }
+
   return NULL;
 }
 
+static const uint32_t *dds_stream_skip_delimited_default (char * __restrict data, const uint32_t * __restrict ops)
+{
+  return dds_stream_skip_default (data, ++ops);
+}
+
+static bool dds_stream_skip_pl_member_default (char * __restrict data, const uint32_t * __restrict ops)
+{
+  uint32_t insn;
+  while ((insn = *ops) != DDS_OP_RTS)
```

Comment:
Methinks this function can be rewritten in a slightly clearer manner: the only successful way out for `dds_stream_skip_pl_member_default` is hitting `DDS_OP_JSR`, and so having a loop that has as its declared termination condition reaching a `DDS_OP_RTS`, that doesn't update the loop variable and which unconditionally `abort`s on reaching the termination condition seems ever so slightly obfuscated.

Maybe I'm missing something, but isn't this function really just:
```
insn = *ops;
assert (DDS_OP (insn) == DDS_OP_JSR);
return dds_stream_skip_pl_member_default (data, ops + DDS_OP_JUMP (insn));
```

Reply:
Yeah, that implementation was definitely incorrect, I made a stripped-down version of `dds_stream_skip_default` and forgot to including updating the `ops` pointer (and the final `abort` shouldn't be there). Because the `ADR` for a PL-member is actually a type on itself delimited by an `RTS`, it can just use `dds_stream_skip_default` for skip-with-default of that member. It slipped through because the type object deserialization only required skipping appendable types, not mutable. In [8409126](https://github.com/eclipse-cyclonedds/cyclonedds/pull/1560/commits/840912631c92ad9229968815364af338d6b96c15) I've added an improved version, but there should be some tests for this, so I'll add these in another commit.